block-storage/src/main/scala/coop/rchain/blockstorage/util/io/RandomAccessIO.scala

```diff
@@ -0,0 +1,55 @@
+package coop.rchain.blockstorage.util.io
+
+import java.io.{FileNotFoundException, IOException, RandomAccessFile}
+import java.nio.file.Path
+
+import cats.data.EitherT
+import cats.effect.Sync
+import cats.implicits._
+import IOError.{IOErr, IOErrT}
+
+import scala.language.higherKinds
+
+final case class RandomAccessIO[F[_]: Sync](file: RandomAccessFile) {
```

Comment:
maybe it'd make sense to encapsulate the `file`, so that receivers of `RandomAccessIO` don't have access to the impure interface underneath

Reply:
Good point